boards/nxp/frdm_mcxn947/frdm_mcxn947.dtsi

```diff
@@ -52,14 +52,26 @@
 			status = "disabled";
 		};
 	};
+
+	nxp_flexio_connector: flexio-connector {
+		compatible = "gpio-nexus";
```

Comment:
We should be defining a custom compatible for the flexio connector, and adding a pin map in the dt binding. See here for an example for the 6 pin FPC touch connector present on NXP RT10xx EVKs: https://github.com/zephyrproject-rtos/zephyr/blob/cce7e9a706ca848f73e4ef2c435668e358b73305/dts/bindings/gpio/nxp,i2c-tsc-fpc.yaml#L10

Also, should we be defining additional pins beyond the INT and RESET? The FlexIO connector is 28 pins in total, and many can be used as GPIOs, right?

Generally the approach taken is to define all pins for the interface in the binding file, and only define the pins that are actually in use as GPIOs in the devicetree for the board. A good example of this would be the 40 pin parallel LCD connector on the RT10xx boards: https://github.com/zephyrproject-rtos/zephyr/blob/cce7e9a706ca848f73e4ef2c435668e358b73305/dts/bindings/gpio/nxp,parallel-lcd-connector.yaml#L13

Reply:
thanks for the suggestion!
duly applied